### PR TITLE
feat: setup a synchronization plan if the resources are in error

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Proposed changes

If the snapshot and the target organization are not in sync, the command will print an URL to the Admin Console so the user can synchronize his organization.
The CLI is not handling any synchronization logic at the moment.

## Testing

- [x] Unit Tests:
    
- [ ] Functionnal Tests:
    
- [ ] Manual Tests:
    

-----
CDX-370

### CODE REVIEW

File: packages/cli/src/commands/org/config/preview.ts [hunk 1]
```diff
@@ -93,7 +75,49 @@ export default class Preview extends Command {
...
+
+    if (snapshot.requiresSynchronization()) {
+      cli.action.start('Synchronization');
+
+      const synchronizationPlanUrl = await this.getSynchronizationPage(
+        snapshot
+      );
+      this.warn(
+        dedent`Some conflicts were detected while comparing changes with the organization.
+        Click on the URL bellow to synchronize your organization before running the command again.
```
Comment (louis-bompart) [inline packages/cli/src/commands/org/config/preview.ts:92]:
```suggestion
        dedent`Some conflicts were detected while comparing changes between the snapshot and the target organization.
        Click on the URL below to synchronize your organization before running the command again.
```

Comment (louis-bompart) [reply packages/cli/src/commands/org/config/preview.ts:92]: Tho I think @jpmarceau expert eye would be welcome here :D

File: packages/cli/src/commands/org/config/preview.ts [hunk 2]
```diff
@@ -93,7 +75,49 @@ export default class Preview extends Command {
     return cfg.organization;
   }
 
+  private async handleInvalidSnapshot(snapshot: Snapshot) {
+    // TODO: CDX-362: handle invalid snashot cases
```
Comment (louis-bompart) [inline packages/cli/src/commands/org/config/preview.ts:79]: can be ignored, 50% suggestion, 50% trolling :P 
```suggestion
    // TODO: CDX-362: handle invalid snapshot cases
```

File: packages/cli/src/lib/snapshot/snapshot.ts [hunk 3]
```diff
@@ -39,10 +41,41 @@ export class Snapshot {
...
+      'snapshot-reports',
+      `${this.latestReport.id}.json`
+    );
+    ensureFileSync(pathToReport);
+    writeJsonSync(pathToReport, this.latestReport, {spaces: 2});
+    return pathToReport;
+  }
+
+  public get latestReport(): ResourceSnapshotsReportModel {
+    if (this.model.reports === undefined || this.model.reports.length === 0) {
```
Comment (louis-bompart) [inline packages/cli/src/lib/snapshot/snapshot.ts:65]: Given what's on line 68:
```suggestion
    if (!Array.isArray(this.model.reports) || this.model.reports.length === 0) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

**Review by louis-bompart — APPROVED** [review]
Code looks good!
I tagged @jpmarceau 'cause I ain't sure about the message tho

File: packages/cli/src/commands/org/config/preview.ts (hunk 1 above)
Comment (jpmarceau) [reply packages/cli/src/commands/org/config/preview.ts:92]: Building on Louis' suggestion:

Comment (jpmarceau) [inline packages/cli/src/commands/org/config/preview.ts:92]:
```suggestion
        Click on the URL bellow to synchronize your snapshot with your organization before running the command again.
```

**Review by jpmarceau — APPROVED** [review]

**Review by olamothe — APPROVED** [review]
💯  !

**Pull request merged**